sdk/storage/inc/common/memory_stream.hpp

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "http/stream.hpp"
+
+#include <vector>
+
+namespace Azure { namespace Storage {
+
+  class MemoryStream : public Azure::Core::Http::BodyStream {
+  public:
+    explicit MemoryStream(const char* data, std::size_t length) : m_data(data), m_length(length) {}
```

Comment:
As talked offline, we need to change the constructor to only accept `const uint8_t*, std::size_t`.